### PR TITLE
Gtk abstraction

### PR DESCRIPTION
**Pull request by Ambrevar**
I've extracted the high-level code from the GTK renderer so that it can be factorer with other renderers (e.g. running hooks).

This saves some 50 lines of code in GTK, which is important considering we are saving them from all future renderers.

### CODE REVIEW

File: source/window.lisp [hunk 1]
```diff
@@ -13,6 +13,25 @@
     (lambda (input)
       (fuzzy-match input windows))))
 
+(declaim (ftype (function (browser)) window-make))
+(defun window-make (browser)
+  "This function must be called by the renderer when a window is deleted."
```
Comment (jmercouris) [inline source/window.lisp]: I assume this is a bit of a typo :-)

**Comment by jmercouris** [comment]
It looks great! Thank you! As I said in my email I would like to use defmethods because there are in fact scenarios with dispatching that may very well be relevant, and since the performance cost is so minuscule, I would prefer it very much.

File: source/window.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/window.lisp]: Oops :p  Good catch!

**Comment by Ambrevar** [comment]
Personally I prefer to use defmethod when I know which are the different implementations.  Otherwise it feels a bit like shooting in the dark.

Anyways, I propose the following: let's merge this and then you can change what you feel like could be a good defmethod (because it's not too clear to me).  What do you think?

**Comment by jmercouris** [comment]
OK, that sounds like a plan to me! Feel free to merge this whenever you like.

**Comment by Ambrevar** [comment]
Merged.

**Pull request closed without merging**